debezium-core/src/main/java/io/debezium/schema/TopicSelector.java

```diff
@@ -158,7 +158,16 @@ public TopicNameCache(DataCollectionTopicNamer<I> delegate) {
 
         @Override
         public String topicNameFor(I id, String prefix, String delimiter) {
-            return topicNames.computeIfAbsent(id, did -> delegate.topicNameFor(did, prefix, delimiter));
+            String topicName = topicNames.get(id);
```

Comment:
@gunnarmorling `get` is not guarded with `synchonized` whci means that `get` and `put` can be executed simultaneously. Are we sure this is a determinist operation?

Reply:
Hum, now I'm unsure, as you bring it up. I really want to avoid going through synchronization though in the normal case. Otherwise we'd build a bottleneck which e.g. may hit us with parallelized snapshots (as already done by the MongoDB connector).

Reply:
So I think you are right, we cannot do it that way. The reason being that `get()` also modifies the structure when using LHM like we do (accessOrder=true).

Reply:
Reworked the last commit to simply use ORM's `BoundedConcurrentHashMap`. As discussed in the chat, we can add a non-threadsafe alternative later on, if actually needed.